app/code/Magento/Catalog/Controller/Adminhtml/Category/Add.php

```diff
@@ -6,49 +6,55 @@
  */
 namespace Magento\Catalog\Controller\Adminhtml\Category;
 
-use Magento\Framework\App\Action\HttpGetActionInterface as HttpGetActionInterface;
+use Magento\Backend\App\Action\Context;
+use Magento\Backend\Model\View\Result\Page;
+use Magento\Framework\Controller\ResultFactory;
+use Magento\Backend\Model\View\Result\Redirect;
+use Magento\Framework\Controller\ResultInterface;
+use Magento\Catalog\Controller\Adminhtml\Category;
+use Magento\Framework\App\Action\HttpGetActionInterface;
 
 /**
  * Class Add Category
  *
  * @package Magento\Catalog\Controller\Adminhtml\Category
  */
-class Add extends \Magento\Catalog\Controller\Adminhtml\Category implements HttpGetActionInterface
+class Add extends Category implements HttpGetActionInterface
 {
     /**
-     * Forward factory for result
+     * Common Factory to create any Result object
      *
-     * @var \Magento\Backend\Model\View\Result\ForwardFactory
+     * @var ResultFactory
      */
-    protected $resultForwardFactory;
+    protected $resultFactory;
 
     /**
      * Add category constructor
      *
-     * @param \Magento\Backend\App\Action\Context $context
-     * @param \Magento\Backend\Model\View\Result\ForwardFactory $resultForwardFactory
+     * @param Context $context
+     * @param ResultFactory $resultFactory
      */
     public function __construct(
-        \Magento\Backend\App\Action\Context $context,
-        \Magento\Backend\Model\View\Result\ForwardFactory $resultForwardFactory
+        Context $context,
+        ResultFactory $resultFactory
```

Comment:
Please revert this change, it's not backward compatible and it's actually not needed

Reply:
Yes, I'll do that. But that's make `$resultForwardFactory` unused code, right?

Reply:
Done, thank you.